src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3178,6 +3199,44 @@ def _get_public_ip_address_allocation(value, sku):
         deployment_result['vmss']['identity'] = _construct_identity_info(identity_scope, identity_role,
                                                                          vmss_info.identity.principal_id,
                                                                          vmss_info.identity.user_assigned_identities)
+    # Guest Attestation Extension and enable System Assigned MSI by default
+    if security_type and enable_vtpm and enable_secure_boot and not disable_integrity_monitoring:
+        client = _compute_client_factory(cmd.cli_ctx)
+        vmss = client.virtual_machine_scale_sets.get(resource_group_name, vmss_name)
+        vmss.virtual_machine_profile.storage_profile.image_reference = None
+        VirtualMachineScaleSetExtension, VirtualMachineScaleSetExtensionProfile = cmd.get_models(
+            'VirtualMachineScaleSetExtension', 'VirtualMachineScaleSetExtensionProfile')
+        if vmss.virtual_machine_profile.storage_profile.os_disk.os_type == 'Linux':
+            publisher = 'Microsoft.Azure.Security.LinuxAttestation'
+        if vmss.virtual_machine_profile.storage_profile.os_disk.os_type == 'Windows':
+            publisher = 'Microsoft.Azure.Security.WindowsAttestation'
+        version = _normalize_extension_version(cmd.cli_ctx, publisher, 'GuestAttestation', None, vmss.location)
+        if cmd.supported_api_version(min_api='2019-07-01', operation_group='virtual_machine_scale_sets'):
+            ext = VirtualMachineScaleSetExtension(name='GuestAttestation',
+                                                  publisher=publisher,
+                                                  type_properties_type='GuestAttestation',
+                                                  protected_settings=None,
+                                                  type_handler_version=version,
+                                                  settings=None,
+                                                  auto_upgrade_minor_version=True,
+                                                  provision_after_extensions=None,
+                                                  enable_automatic_upgrade=None)
+        else:
+            ext = VirtualMachineScaleSetExtension(name='GuestAttestation',
+                                                  publisher=publisher,
+                                                  type='GuestAttestation',
+                                                  protected_settings=None,
+                                                  type_handler_version=version,
+                                                  settings=None,
+                                                  auto_upgrade_minor_version=True,
+                                                  provision_after_extensions=None,
+                                                  enable_automatic_upgrade=None)
```

Comment:
`type_properties_type` is supported after api-version `2019-07-01`, right?
Could we simplify this code, such as:
```
        ext = VirtualMachineScaleSetExtension(name='GuestAttestation',
                                              publisher=publisher,
                                              type='GuestAttestation',
                                              protected_settings=None,
                                              type_handler_version=version,
                                              settings=None,
                                              auto_upgrade_minor_version=True,
                                              provision_after_extensions=None,
                                              enable_automatic_upgrade=None)
        if cmd.supported_api_version(min_api='2019-07-01', operation_group='virtual_machine_scale_sets'):
            ext.type_properties_type='GuestAttestation'
```

Reply:
In addition, please remember to run the tests for the old profiles

Reply:
The version of Trusted Launch includes it, so remove。